app/controllers/api_controller/virtual_templates.rb

```diff
@@ -0,0 +1,20 @@
+class ApiController
+  module VirtualTemplates
+    def create_resource_virtual_templates(type, _id, data)
+      validate_vendor_present(data)
+      klass = collection_class(type)
+      virtual_template = klass.create(data)
+      if virtual_template.invalid?
+        raise BadRequestError,
+              "Failed to create a new virtual template - #{virtual_template.errors.full_messages.join(', ')}"
+      end
+      virtual_template
+    end
+
+    private
+
+    def validate_vendor_present(data)
+      raise BadRequestError, 'Must specify a vendor for creating a Virtual Template' unless data['vendor']
```

Comment:
@chriskacerguis could you update validation to make sure id and href are not in the data.